### ISSUE TITLE
Improving Dataverse's Schema.org JSON-LD schema to enable author names display in Google Dataset Search's

### ISSUE THREAD

**Issue by chewsw** [IQSS/dataverse#5029]
Author names are missing from the dataset records that Google Dataset Search indexed from Dataverse.

The JSON-LD schema for authors should be updated to "@type": "Person" or other appropriate types to differentiate between organization as authors or individuals as authors.

**Comment by pdurbin** [comment]
@chewsw thanks for opening this issue. Here's the thread from the dataverse-users mailing list: https://groups.google.com/d/msg/dataverse-community/TlQPNI3Ip2E/srLf29aSBAAJ

Originally we had `"@type": "Person"` in the JSON-LD output (in development, before release) but in Dataverse it's possible to have organizations as authors ("Gallup Organization", "Geological Survey (U.S.)", etc.) so we took it out. Please see discussion in these two places:

- https://github.com/IQSS/dataverse/issues/2243#issuecomment-345363935
- eec1163

**Comment by jggautier** [comment]
Maybe outlining some more details would help with estimation:
- This will be required metadata that the author can change from the UI or API
- We need a plan for how installations can add creator types to existing datasets

**Comment by jggautier** [comment]
I'm starting to think that Google Dataset Search prefers the `creator` property (as opposed to the `author` property that Dataverse uses). For every dataset landing page I've found in Google Dataset Search where Google shows the author, the `creator` property is used instead of the `author` property.

Unless someone finds something different, I'd propose that Dataverse uses the `creator` property instead:

```
"creator": [
    {
      "affiliation": "affiliation",
      "@type": "Person",
      "name": "Lname, Fname"
    },
    {
      "affiliation": "affiliation",
      "@type": "Organization",
      "name": "Org name"
    }
```

**Comment by pdurbin** [comment]
Today during sprint planning @jggautier explained his hunch on how switching from `author` to `creator` might help. This was while discussing #4371.

**Comment by jggautier** [comment]
I [asked for clarification](https://productforums.google.com/forum/#!topic/webmasters/Ix1PXcY9IHc) in the structured data section of Google's webmaster forum.

**Comment by chewsw** [comment]
Hi Julian,

Thanks very much for following up on this. Changing the "author" property
to "creator" seems like a good idea. Let's see how Google will respond to
this!

**Comment by jggautier** [comment]
Author names are showing up on some but not all Google Dataset Search pages for datasets in Dataverse repositories, like [this page for a dataset from the Texas Data Repository (TDR)](https://toolbox.google.com/datasetsearch/search?query=Texas%20Data%20Repository&docid=lrWTeGixyRwCdiXKAAAAAA%3D%3D), and [this page for a dataset from Harvard Dataverse](https://toolbox.google.com/datasetsearch/search?query=Harvard%20Dataverse&docid=93ZNviSVKfN13dXBAAAAAA%3D%3D). But those pages also say metadata is coming from DataCite, which publishes its own schema.org metadata and uses only the "author" property, but includes its guessed `@type` (e.g. [this schema.org metadata from DataCite for that TDR dataset](https://api.datacite.org/dois/application/vnd.schemaorg.ld+json/10.18738/t8/xyqvnu)). From what I can tell so far, every Google Dataset Search page for datasets from a Dataverse repository includes author names only when the "dataset provided by" includes DataCite.

Harvard Dataverse upgraded to Dataverse 4.10.1 two days ago (Jan 8), which includes adding the "creator" property to the schema.org metadata. Once Google starts indexing more recently published datasets, we can see if authors are displayed on Google Dataset Search pages (especially when it isn't also isn't using DataCite's schema.org metadata).

**Comment by jggautier** [comment]
Just an update: Datasets published in Harvard Dataverse after Jan 7, with the updated Schema.org metadata, are showing up in Google Dataset Search without the author names (like [this one](https://toolbox.google.com/datasetsearch/search?query=%22Using%20Driver%20State%20Detection%20in%20Automated%20Vehicles%22&docid=fjdVSeY7a3kirVamAAAAAA%3D%3D) and [this one](https://toolbox.google.com/datasetsearch/search?query=%22High-field%20side%20scrape-off%20layer%20investigation%3A%20plasma%20profiles%20and%20impurity%20screening%20behavior%20in%20near-double-null%20configurations%22&docid=2%2BD5%2B%2B95sTbDE6qvAAAAAA%3D%3D)). I think we can rule out any preference for "author" versus "creator" elements.

**Comment by pdurbin** [comment]
@jggautier bummer. Does that mean we should try adding `"@type": "Person"`? As indicated above, the Dataverse UI/API would need to allow dataset authors to choose between a person and an organization.

**Comment by jggautier** [comment]
That or we could do what [@mfenner wrote](https://github.com/IQSS/dataverse/issues/2243#issuecomment-358615313) in #2243 that DataCite does, which is basically guess (with >90% accuracy).

**Comment by Venki18** [comment]
@pdurbin and @jggautier thank you for looking into this problem. I thought I will add my findings here if that will help to make the changes in the next version. 

What I discovered using the Structured dataset test tool is for NTU datasets the author fields show as Thing in the test tool. If I change the "@type": "Person" for one of the authors the tool doesn't show any error. I think we must include the "@type": "Person" for all the authors in the ld+json in the script. Then google dataset results page will show the author names under Person. 

![image](https://user-images.githubusercontent.com/19611506/58938894-780ec480-87a8-11e9-978f-fb4853993198.png)

I see from an example of GBIF—the Global Biodiversity Information Facility dataset record in Google dataset results page display author names as "@type": "Person".

[![image](https://user-images.githubusercontent.com/19611506/58939616-2ebf7480-87aa-11e9-9878-dcc20eae7d9a.png)](https://doi.org/10.15468/sdyzw5)

Please refer to the screenshot below
![image](https://user-images.githubusercontent.com/19611506/58940143-53681c00-87ab-11e9-8193-9b04d61b094f.png)

I think we need to add "@type": "Person" in Dataverse to show the author name in Google dataset page.

**Comment by pdurbin** [comment]
@Venki18 hi! Yes, the "every person and organization is a Thing" problem is well known to us, unfortunately.

I've been hoping we can use some new code added by @fcadili in pull request #4664 to pass in a string that could either be a person or an organization and the code will tell use which it is.

I haven't studied the code yet but here's a test he wrote that shows the code figuring out if a string is for an organization or a person, for example:

https://github.com/IQSS/dataverse/blob/v4.14/src/test/java/edu/harvard/iq/dataverse/export/OrganizationsTest.java

![Screenshot from 2019-06-05 06-23-00](https://user-images.githubusercontent.com/21006/58949426-67cff880-875a-11e9-96f4-e60193181040.png)

**Comment by Venki18** [comment]
@pdurbin thank you for the quick reply. May I know how does the export for TermsOfUse work?  We have been using CC-BY-NC instead of CC0 and we have changed the necessary text in Bundle.properties file.  But we are using the code CC0 as it is. Hence when you guys export to ld+json format it is exported as CC0. Hence for all our datasets with waiver terms the code exports what is entered in the additional text box. For default CC0 it is taken as it is. 
Is there any way to show CC-BY-NC?

**Comment by pdurbin** [comment]
@Venki18 I'm not sure but let me at least give you and @Thanh-Thanh and others some pointers to the code:

https://github.com/IQSS/dataverse/blob/v4.14/src/main/java/edu/harvard/iq/dataverse/DatasetVersion.java#L1777

It looks like if CC0 isn't specified the code will put in the free form text the user entered as an alternative to CC0.

This is somewhat off topic for this issue, of course, but I hope this helps! :smile: Please feel free to create as many issues as we need!

**Comment by pdurbin** [comment]
@Venki18 also, if you're interested @rigelk and I are talking about Schema.org JSON-LD, especially in relation to ActivityPub (#5883) in chat. You can catch up on the conversation at http://irclog.iq.harvard.edu/dataverse/2019-06-05

**Comment by adam3smith** [comment]
Just to register our interest in fixing this, using the Datacite strategy of guessing. Would you take a fix along those lines as a PR? Otherwise we'd address it locally -- not having authors in the Google Dataset search is a bummer.

**Comment by jggautier** [comment]
I think we were worried about how well [the Datacite strategy of guessing the name type] (https://github.com/IQSS/dataverse/issues/2243#issuecomment-358615313) (used in the Dataverse software's OpenAIRE metadata export) would work for all Dataverse installations.

These are some or hopefully all of the next steps proposed already in another GitHub issue (which I can't find right now), some by @qqmyers:
- Test how well the algorithm for guessing author name types works for different installations, especially those with different types of names, and improve the algorithm if needed before using it in other metadata exports like the Schema.org export
- Let installations decide whether or not to use the algorithm for any of their metadata exports.
- Use the algorithm for datasets already created in an installation (and ask depositors to indicate the type of author name being entered)

I think a combination of these things could be done, like testing the algorithm and adding a way for installations to say that they don't want to use it. In this case I'd advocate for providing installations with guidance about how to evaluate the accuracy of the algorithm. If GUI work could be done to let depositors choose the name type when they enter names, installations could also decide not to use the algorithm to add "type" metadata to the author/creator names of datasets already created in their installations.

But one blocker might be a lack of resources (people and time) to review the algorithm and design and test GUI changes. Maybe a review of the algorithm's accuracy would be less resource intensive if it was distributed among a representative sample of Dataverse installations, like those with names from different cultures in their metadata. Each installation could test the accuracy of their own metadata, report back, and the community could make a decision about what to do then.

The algorithm is being used to determine the name type for names entered in other fields, like the Contact Person field, but I'm also assuming that we would prioritize identifying the types of names in the authors/creator fields because Google Dataset Search usually displays those name but isn't when there's no name type in the Schema.org metadata.

**Comment by adam3smith** [comment]
Thanks for responding so quickly!

> I think we were worried about how well [the Datacite strategy of guessing the name type] (https://github.com/IQSS/dataverse/issues/2243#issuecomment-358615313) (used in the Dataverse software's OpenAIRE metadata export) would work for all Dataverse installations.

I understand this, but I'm wondering whether this is the right trade-off:
- With the current behavior, we're generating invalid JSON-LD and having our data presented at Google without proper attribution to their creators 
- With a worst-case scenario of implementing the fix and "guessing" working for only half the data (a _highly_ unlikely scenario), we'd fix the invalid JSON-LD, have our holding show up properly in google, with the main downside that some people would be categorized as organizations and vice versa. The main effect of this would be, I think, that some auto-generated citations would look a bit off.

Am I overlooking a massive cost to incorrectly labelling a creator?
In other words, even if the algo would work poorly in some installations -- is there any scenario where it's worse than the status quo? And if it isn't is there any way in which implementing a "guessing" fix would create a form of technology lock-in that we want to avoid? Otherwise, I don't really see the case against this.

**Comment by adam3smith** [comment]
> I don't think there's any technical lock-in, if I get your meaning. I think it would be easy to improve this over time, right? 

Right, that's what I mean -- I don't think this would make implementing either better algorithmic solutions or a manual UI for type selection in the future harder.

> The lock-in might be more behavioral and I'm probably suggesting that we front-load research because I always worry that the effectiveness of the changes that make it into each Dataverse release won't be evaluated soon enough, if at all. By the time we see that something needs to be fixed, the problem could be much more difficult to fix.

That's fair. I guess beyond better algos (which I expect would be hard) the relevant question is whether, and to what extent, there is the need to allow author-type selection in the GUI, which would obviously be more precise in theory, but is also another UI feature and itself prone to user error, especially for large, self-curated repositories like Harvard DV.
How would you do the research on this? Just sample and look manually? At QDR we can obviously look at our entire holdings but I also know our metadata well enough to be pretty certain that even a naive algo without first-name matching would get us to 100%.

**Comment by adam3smith** [comment]
>  So maybe that's enough to be confident that the algorithm would be accurate enough, and instead we could consider here or later on why someone would want to correct a guess that the algorithm made and how they might be able to.

Great, yes, let's make that the plan

**Comment by jggautier** [comment]
@adam3smith, about planning to review why someone would want to correct a guess that the algorithm made and how they might be able to, I wanted to clarify that I think this should be done before the algorithm is used to add metadata to other exports, like the Schema.org export.

What do you think?

Very early in the Schema.org conversation and development work, I asked a liaison at Google why Google Dataset Search would insist on a nameType, but I thought the answer was vague and left a lot of room for speculation (like, their knowledge graph could take advantage of knowing more about dataset authors), which at the time I think made it easy for us to feel okay with ignoring the nameType error we kept seeing when checking the Schema.org exports (until we started noticing that if the dataset didn't have a person or organization nameType, the author names wouldn't show up in Google Dataset Search).

But since you mentioned that the algorithm guessing wrong might result in an off-looking citation, and Martin Fenner wrote earlier that the algorithm is necessary for generating correct citations, maybe we could start by assuming that the only way that a depositor or curator would notice the wrong guess is if they saw a citation of the dataset that was off. They could always just correct the autogenerated citation themselves, right? I remember having to do this often when managing citations in Zotero. But what are the chances that this becomes a big inconvenience and someone wants a curator or repository manager to correct the generated citations? In your experience, is the risk of that happening so low that we don't need to consider how a depositor or curator could correct a wrong guess? You mentioned that this won't happen in QDR, but might it for other Dataverse repositories?

**Comment by adam3smith** [comment]
> But what are the chances that this becomes a big inconvenience and someone wants a curator or repository manager to correct the generated citations?

I think for personal authors this is a real issue and it comes up with poor metadata in Zotero a fair amount, *but*:
Given the way the algo works, this is easy to do for person authors and we don't need any additional functionality for this (it's also why I think it's irrelevant at QDR): As long as you add your name as "Lastname, Firstname" _or_ you have an ORCID in the metadata, you're _always_ going to be a person. Lastname, Firstname is, of course, already explicitly recommended by Dataverse as the mode of entry for just this reason.

The only scenario for which we'd therefore need new functionality is an organization author that gets misclassified by the algo, either because they have a comma in their name or because they  have a first word that looks like a first name. This, I think, is going to be incredibly rare and I'd argue not worth considering. The chances of it happening could be further reduced if the last step of the algorithm is skipped, i.e. we treat all names without a comma as organizations, but I don't think that's a good idea given the trade-offs of missing a large number of 'incorrectly' added persons.

**Comment by pdurbin** [comment]
This PR is related, at least:

- #9089

Might fix it? Or go a long way? I'm not sure! 😄

**Comment by qqmyers** [comment]
It probably does close it but there's a lot of discussion above. Perhaps whatever's left, if anything, can be a new issue?

**Comment by pdurbin** [comment]
@qqmyers fantastic. I just marked that PR (#9089) as closing this issue.

To anyone reading this, please look through that pull request. If there are any remaining issues after it is merged, please open a fresh issue. Thanks! 🚀

**Comment by adam3smith** [comment]
Here's how things look on our end 😄 
![image](https://user-images.githubusercontent.com/128986/214355607-93ffa583-bea4-4a61-a526-c7946497d20c.png)

**Comment by jggautier** [comment]
Thanks @qqmyers for always helping make sure things don't fall through the cracks!

The two things I see in this issue that aren't addressed in PR https://github.com/IQSS/dataverse/pull/9089 are:

- The licensing issues that @Venki18 [mentioned in June 2019,](https://github.com/IQSS/dataverse/issues/5029#issuecomment-499038090) which may have been resolved in v5.10 with the PR https://github.com/IQSS/dataverse/pull/7920. In the installation that @Venki18 helps manages, license metadata in the Schema.org exports seems to be represented well, such as the CC BY-NC 4.0 dataset at https://doi.org/10.21979/N9/OXUQQA and its Schema.org export at https://researchdata.ntu.edu.sg/api/datasets/export?exporter=schema.org&persistentId=doi:10.21979/N9/OXUQQA.
- Determining how well Dataverse guesses the type of author, using that algorithm. From my conversation with Sebastian, it makes sense to me for a review to pay more attention to how well the algorithm identifies organizational authors as organizations and how well it identifies people authors as people when the author's metadata doesn't include an ORCID and isn't in the lastname, first name format (or includes multiple commas?). I can open a GitHub issue about this review, which I think we've taken to calling spikes.
